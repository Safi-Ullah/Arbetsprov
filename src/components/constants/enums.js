export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};