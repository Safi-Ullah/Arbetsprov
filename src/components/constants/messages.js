export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};