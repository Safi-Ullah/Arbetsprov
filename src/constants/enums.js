export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};