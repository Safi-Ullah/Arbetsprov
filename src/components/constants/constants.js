import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};