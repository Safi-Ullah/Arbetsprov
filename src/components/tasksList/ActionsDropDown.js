import React from 'react';
import PropTypes from 'prop-types';
import { Enums, TASK_ACTION_HANDLERS } from '../constants';
import { getKeyFromValue } from '../../utils';

export const ActionsDropDown = props => {

    const { taskNextStatus, changeStatus, task } = props;

    const taskActions = {
        'Edit': Enums.TASK_ACTIONS.edit
    };

    return <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
        {
            Object.keys(taskActions).map(action => <button
                key={taskActions[action]}
                onClick={() => props[TASK_ACTION_HANDLERS[taskActions[action]]](task.id)}
                className="dropdown-item">
                { action }
            </button>)
        }
        {
            taskNextStatus.length
                ? <React.Fragment>
                    <div className="dropdown-divider" />
                    <span className="dropdown-subheading">Change Status</span>
                </React.Fragment>
                : null
        }
        {
            taskNextStatus.map(status => <button
                key={status}
                onClick={() => changeStatus(task.id, status)}
                disabled={
                    (status === Enums.TASK_STATUS.active && !task.assignee) ||
                    (status === Enums.TASK_STATUS.completed && task.status !== Enums.TASK_STATUS.active)
                }
                className="dropdown-item text-capitalize">
                { getKeyFromValue(Enums.TASK_STATUS, status) }
            </button>)
        }
    </div>;
};

ActionsDropDown.propTypes = {
    task: PropTypes.object.isRequired,
    taskNextStatus: PropTypes.array.isRequired,
    changeStatus: PropTypes.func.isRequired
};