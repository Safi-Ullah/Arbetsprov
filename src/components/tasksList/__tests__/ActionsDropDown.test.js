import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ActionsDropDown } from '../ActionsDropDown';
import { Enums } from '../../../constants';

const props = {
    task: {
        id: 1,
        title: 'test',
        desc: 'test',
        complexity: '0',
        assignee: 'test',
        status: Enums.TASK_STATUS.created
    },
    taskNextStatus: [],
    changeStatus: jest.fn()
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('show task dropdown actions', () => {
        const tree = TestRenderer.create(<ActionsDropDown {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});