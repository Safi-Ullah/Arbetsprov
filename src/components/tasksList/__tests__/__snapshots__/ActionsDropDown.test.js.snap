// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test show task dropdown actions 1`] = `
<div
  aria-labelledby="dropdownMenuButton"
  className="dropdown-menu"
>
  <button
    className="dropdown-item"
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;
